docs/src/main/asciidoc/security-authorize-web-endpoints-reference.adoc

```diff
@@ -1055,8 +1055,79 @@ Because `MediaLibrary` is the `TvLibrary` class parent, a user with the `admin`
 
 CAUTION: Annotation-based permissions do not work with custom xref:security-customization.adoc#jaxrs-security-context[Jakarta REST SecurityContexts] because there are no permissions in `jakarta.ws.rs.core.SecurityContext`.
 
+[[permission-checker]]
+==== Create permission checkers
+
+By default, `SecurityIdentity` must be augmented with permissions which can be used to check if this identity passes `@PermissionAllowed` authorization restrictions.
+Alternatively, you can use a `@PermissionChecker` annotation to mark any CDI bean method as a permission checker.
+For example, a permission checker can be created like this:
+
+[source,java]
+----
+package org.acme.security.rest.resource;
+
+import io.quarkus.security.PermissionChecker;
+import io.quarkus.security.PermissionsAllowed;
+import io.quarkus.security.identity.SecurityIdentity;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.Path;
+
+import org.jboss.resteasy.reactive.RestForm;
+import org.jboss.resteasy.reactive.RestPath;
+
+@Path("/project")
+public class ProjectResource {
+
+    @PermissionsAllowed("rename-project")
+    @POST
+    public void renameProject(@RestPath String projectName, @RestForm String newName) {
+        Project project = Project.findByName(projectName);
+        project.name = newName;
+    }
+
+    @PermissionChecker("rename-project") <2>
```

Comment:
Does it has to be in the same bean? I would prefer an example with 2 different beans.

What about scope BTW (when it's 2 different beans)?

Reply:
> Does it has to be in the same bean? I would prefer an example with 2 different beans.

It doesn't need to be same bean. Couple of lines below is an example that shows that.  I can drop this one because it is not important for me at all. When I write docs, I try to show all that is possible and this just looked nice in my eyes. If you require dropping this, I will do that.

> What about scope BTW (when it's 2 different beans)?

Before security checks are started (that is not specific for this method), CDI request scope is activated. It is same request scope.

Reply:
Ok, I think it should explain that it could be in a different bean (and point to the next example, which I've seen 5 min later). 

About the scope, it should also be mentioned that you can rely on the request scope. 

BTW, is the bean providing the checker necessarily application scoped or singleton? What about dependent, or request scoped?

Reply:
> Ok, I think it should explain that it could be in a different bean (and point to the next example, which I've seen 5 min later).
> About the scope, it should also be mentioned that you can rely on the request scope.

It makes sense, I'll do it.

> BTW, is the bean providing the checker necessarily application scoped or singleton? What about dependent, or request scoped?

There is validation in place, it must be either `Singleton`, `ApplicationScoped` or `RequestScoped`. I decided against `Dependent` because `ArcContainer` Javadoc says  that you must explicitly destroy dependent beans when they are accessed programmatically and there reliable hook when `permission` is not going to be used anymore. I think we could add it later if there is sound reason.

Reply:
I'll mention that in docs as well.